### ISSUE TITLE
Regexes on Unicode strings don't work correctly in 5.19.7

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13571]
Migrated from [rt.perl.org#121144](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=121144) (status was 'resolved')

Searchable as RT121144$

**Comment by p5pRT** [comment]
### From alex.karelas@gmail.com

#### Created by alex.karelas@gmail.com




The following very simple program makes $t2 equals to undef\, whereas it 
should equal the greek capital Sigma character\.
On Perl 5\.14 \(Ubuntu\) $t2 equals greek capital Sigma\. But in Perl 5\.19\.7 
it equals undef\. Therefore\, I think it is a bug\.

\#\!/usr/bin/perl

use strict;
use warnings;
use utf8;

use Data&#8203;::Dumper;

my $s = 'Σ=Π';

my \($t1\) = $s =~ /^\(\\S\+\)/g;
my \($t2\) = $s =~ /^\(\\S\+\)=/g;

print Dumper\($t1\, $t2\);

<details><summary>Perl Info</summary>

```

Flags:
category=core
severity=critical

Site configuration information for perl 5.19.7:

Configured by karjala at Sun Feb 2 14:56:41 EET 2014.

Summary of my perl5 (revision 5 version 19 subversion 7) configuration:

Platform:
osname=linux, osvers=3.5.0-45-generic, archname=x86_64-linux
uname='linux test-vm 3.5.0-45-generic #68-ubuntu smp mon dec 2 21:58:52 
utc 2013 x86_64 x86_64 x86_64 gnulinux '
config_args='-de -Dprefix=/home/karjala/perl5/perlbrew/perls/perl-5.19.7 
-Dusedevel 
-Aeval:scriptdir=/home/karjala/perl5/perlbrew/perls/perl-5.19.7/bin'
hint=recommended, useposix=true, d_sigaction=define
useithreads=undef, usemultiplicity=undef
useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
use64bitint=define, use64bitall=define, uselongdouble=undef
usemymalloc=n, bincompat5005=undef
Compiler:
cc='cc', ccflags ='-fno-strict-aliasing -pipe -fstack-protector 
-I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
optimize='-O2',
cppflags='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
ccversion='', gccversion='4.7.2', gccosandvers=''
intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', 
lseeksize=8
alignbytes=8, prototype=define
Linker and Libraries:
ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
libpth=/usr/local/lib /lib/x86_64-linux-gnu /lib/../lib 
/usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib /usr/lib
libs=-lnsl -lgdbm -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
libc=, so=so, useshrplib=false, libperl=libperl.a
gnulibc_version='2.15'
Dynamic Linking:
dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib 
-fstack-protector'



@INC for perl 5.19.7:
/home/karjala/perl5/lib/perl5/x86_64-linux-gnu-thread-multi
/home/karjala/perl5/lib/perl5
/home/karjala/perl5/perlbrew/perls/perl-5.19.7/lib/site_perl/5.19.7/x86_64-linux
/home/karjala/perl5/perlbrew/perls/perl-5.19.7/lib/site_perl/5.19.7
/home/karjala/perl5/perlbrew/perls/perl-5.19.7/lib/5.19.7/x86_64-linux
/home/karjala/perl5/perlbrew/perls/perl-5.19.7/lib/5.19.7
.


Environment for perl 5.19.7:
HOME=/home/karjala
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_ADDRESS=en_IE.UTF-8
LC_IDENTIFICATION=en_IE.UTF-8
LC_MEASUREMENT=en_IE.UTF-8
LC_MONETARY=en_IE.UTF-8
LC_NAME=en_IE.UTF-8
LC_NUMERIC=en_IE.UTF-8
LC_PAPER=en_IE.UTF-8
LC_TELEPHONE=en_IE.UTF-8
LC_TIME=en_IE.UTF-8
LD_LIBRARY_PATH (unset)
LOGDIR (unset)
PATH=/home/karjala/perl5/perlbrew/bin:/home/karjala/perl5/perlbrew/perls/perl-5.19.7/bin:/home/karjala/.rvm/gems/ruby-2.0.0-p353/bin:/home/karjala/.rvm/gems/ruby-2.0.0-p353@global/bin:/home/karjala/.rvm/rubies/ruby-2.0.0-p353/bin:/home/karjala/Komodo-Edit-8/bin:/home/karjala/perl5/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/karjala/.rvm/bin:/home/karjala/.rvm/bin
PERL5LIB=/home/karjala/perl5/lib/perl5/x86_64-linux-gnu-thread-multi:/home/karjala/perl5/lib/perl5
PERLBREW_BASHRC_VERSION=0.67
PERLBREW_HOME=/home/karjala/.perlbrew
PERLBREW_MANPATH=/home/karjala/perl5/perlbrew/perls/perl-5.19.7/man
PERLBREW_PATH=/home/karjala/perl5/perlbrew/bin:/home/karjala/perl5/perlbrew/perls/perl-5.19.7/bin
PERLBREW_PERL=perl-5.19.7
PERLBREW_ROOT=/home/karjala/perl5/perlbrew
PERLBREW_VERSION=0.67
PERL_BADLANG (unset)
PERL_LOCAL_LIB_ROOT=/home/karjala/perl5
PERL_MB_OPT=--install_base /home/karjala/perl5
PERL_MM_OPT=INSTALL_BASE=/home/karjala/perl5
SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @karenetheridge

Confirmed\, on 5\.18\.2\, this prints&#8203;:

&nbsp;    $VAR1 = "\\x\{3a3\}=\\x\{3a0\}";
&nbsp;    $VAR2 = "\\x\{3a3\}";

5\.19\.0 through 5\.19\.4 also behave the same way\, with behaviour changing in 5\.19\.5\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @khwilliamson

On 02/02/2014 11&#8203;:11 AM\, Karen Etheridge via RT wrote&#8203;:
> Confirmed\, on 5\.18\.2\, this prints&#8203;:
>
>      $VAR1 = "\\x\{3a3\}=\\x\{3a0\}";
>      $VAR2 = "\\x\{3a3\}";
>
> 5\.19\.0 through 5\.19\.4 also behave the same way\, with behaviour changing in 5\.19\.5\.
>
> \-\-\-
> via perlbug&#8203;:  queue&#8203;: perl5 status&#8203;: new
> https://rt-archive.perl.org/perl5/Ticket/Display.html?id=121144
>

Unlike what I expected\, a bisect blames my commit\, so I'll look into it

77ebeeba5f5d47be96b3cef485cb98b1c4980510 is the first bad commit
commit 77ebeeba5f5d47be96b3cef485cb98b1c4980510
Author&#8203;: Karl Williamson \<public@&#8203;khwilliamson\.com>
Date&#8203;:   Mon Sep 30 14&#8203;:06&#8203;:23 2013 \-0600

&nbsp;     PATCH&#8203;: \[perl \#120041\] regcomp\.c missing parens and broken STCLASS

&nbsp;     This was caused by a problem in commit
&nbsp;     43a64b8b430604bd4d76fd256a5babdccaf0ab2b
&nbsp;     which was masked by problems in commit
&nbsp;     a0dd42312a1f26356d2fdf49656e45b77c2cefb5\.

&nbsp;     The first commit deleted a line it shouldn't have\.  I thought that a
&nbsp;     variable wasn't modified before reaching that line\, but it turns out it
&nbsp;     was because another variable is set to point to its address\, and was
&nbsp;     modified inside a subroutine call\.  The other two lines deleted in the
&nbsp;     first commit are ok to have deleted\.

&nbsp;     The second commit masked the problems of the first by omitting some
&nbsp;     necessary grouping parentheses\, which caused things to not work as
&nbsp;     intended\, and we didn't have a good test case in our suite to find this
&nbsp;     problem\.  \(Thanks to Lukas Mai for spotting the problem and 
submitting a
&nbsp;     test case\.\)

**Comment by p5pRT** [comment]
### From @khwilliamson

This is now fixed by 33e0eee4a7d8eddf7d34cbaa10065e77c8410b40
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @ilmari

"Karl Williamson via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:

> This is now fixed by 33e0eee4a7d8eddf7d34cbaa10065e77c8410b40

That commit does not exist\. Did you mean this one?

&nbsp;    commit 3c075feabc1b777553a63a5a7d87ef482f2e3d49
&nbsp;    Author&#8203;: Karl Williamson \<public@&#8203;khwilliamson\.com>
&nbsp;    Date&#8203;:   2014\-02\-02 22&#8203;:30&#8203;:10 \-0700

&nbsp;        PATCH \[perl \#121144\]&#8203;: \\S\, \\W\, etc fail for above ASCII

\-\- 
\- Twitter seems more influential \[than blogs\] in the 'gets reported in
&nbsp;  the mainstream press' sense at least\.               \- Matt McLeod
\- That'd be because the content of a tweet is easier to condense down
&nbsp;  to a mainstream media article\.                      \- Calle Dybedahl

**Comment by p5pRT** [comment]
### From @khwilliamson

On 02/03/2014 04&#8203;:36 PM\, Dagfinn Ilmari Mannsåker wrote&#8203;:
> "Karl Williamson via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:
>
>> This is now fixed by 33e0eee4a7d8eddf7d34cbaa10065e77c8410b40
>
> That commit does not exist\. Did you mean this one?
>
>      commit 3c075feabc1b777553a63a5a7d87ef482f2e3d49
>      Author&#8203;: Karl Williamson \<public@&#8203;khwilliamson\.com>
>      Date&#8203;:   2014\-02\-02 22&#8203;:30&#8203;:10 \-0700
>
>          PATCH \[perl \#121144\]&#8203;: \\S\, \\W\, etc fail for above ASCII
>

Yes\, thanks for finding that; I don't know how I made that mistake\.  I 
try to only use a commit number after it has successfully been pushed to 
blead\.
